### DISCUSSION TITLE
Manifest file overrides version with overlay port version mismatch

### DISCUSSION THREAD

**Discussion by Te93x** [microsoft/vcpkg#36252] [Q&A] [answered]
The dependency version pinned in `vcpkg.json` manifest file doesn't seem to enforce the same version (throw error) if the dependency is an overlay port.

For example, I have an an overlay port `ports\private-package\vcpkg.json`:

```json
{
  "name": "private-package",
  "version": "1.5.0",
  "dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
}
```

And in root `vcpkg.json` I pin the `private-package` version to any version other than the actual version in the overlay port:
```json
{
  "name": "project",
  "version": "1.0.0",
  "dependencies": ["private-package"],
  "overrides": [{ "name": "private-package", "version": "2.0.0" }],
  "builtin-baseline": "98f22593167d1d20a433d9ad09aa40ccd2552749"
}
```

My project continues to build with `private-package:x64-windows@1.5.0`, is this the intended behavior for vcpkg to ignore versions in overrides if it's an overlay port? or a bug?

**Comment by dg0yt** [comment, marked as answer]
Overlay ports have priority. If you want to use versions from alternative sources with manifest mode, you should look at registries.
